x/rollup/keeper/msg_server_test.go

```diff
@@ -0,0 +1,236 @@
+package keeper_test
+
+import (
+	"encoding/json"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	gethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/golang/mock/gomock"
+	"github.com/polymerdao/monomer/testutils"
+	"github.com/polymerdao/monomer/x/rollup/keeper"
+	"github.com/polymerdao/monomer/x/rollup/types"
+)
+
+func (s *KeeperTestSuite) TestApplyL1Txs() {
+	l1AttributesTx, depositTx, cosmosEthTx := testutils.GenerateEthTxs(s.T())
+	// The only constraint for a contract creation tx is that it must be a non-system DepositTx with no To field
+	contractCreationTx := gethtypes.NewTx(&gethtypes.DepositTx{})
+
+	l1AttributesTxBz, err := l1AttributesTx.MarshalBinary()
+	s.Require().NoError(err)
+	depositTxBz, err := depositTx.MarshalBinary()
+	s.Require().NoError(err)
+	cosmosEthTxBz, err := cosmosEthTx.MarshalBinary()
+	s.Require().NoError(err)
+	contractCreationTxBz, err := contractCreationTx.MarshalBinary()
+	s.Require().NoError(err)
+	invalidTxBz := []byte("invalid tx bytes")
+
+	tests := map[string]struct {
+		msg         *types.MsgApplyL1Txs
```

Comment:
nit: might make the test cases simpler to read if they are passed in as `[][]byte`, and then the `MsgApplyL1Txs` is built in the test itself.